docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,819 @@
+		txOut.Value = perRecipientValue
+	}
+
+	return tx, nil
+}
+```
+<1> These two come from `fundingInfo().utxoOutpoint`, which has the transaction
+    id as the first 32 little-endian bytes and the previous output index as
+    its last 4 little-endian bytes.
+<2> Comes from `fundingInfo().utxoValue`, a little-endian unsigned int.
+<3> Derived as part of the <<signing-protocol>>.
+<4> e.g. &chaincfg.MainNetParams or TestNet3Params
+
+
+[[sighash-construction]]
+==== Construction of Sighash for Signing
+
+The sighash is constructed as the double-SHA256 of the preimage. The preimage
+is defined in BIP143. Fortunately, once the dummy unsigned transaction is
+available, btcutil provides the tooling to build the sighash. The sample
+function call below shows how to construct the sighash once an unsigned
+transaction is available from above the function in
+<<transaction-construction>>. This sighash is the data then signed by the
+signer group to produce a valid transaction (see the section on
+<<final-transaction-construction>> for the rest of the process once the
+signature has been computed):
+
+```go
+// import "github.com/btcsuite/btcd/txscript"
+
+// unsignedTransaction := ConstructUnsignedTransaction(...)
+sighashBytes, err := txscript.CalcWitnessSigHash(
+  previousOutputScriptCode, // from PublicKeyToP2WPKHScriptCode
+  txscript.NewTxSigHashes(unsignedTransaction),
+  txscript.SigHashAll,
+  unsignedTransaction,
+  0, // sighash for the first input; there is only one input anyway
+  previousOutputValue,
+)
+```
+
+[[final-transaction-construction]]
+==== Construction of Final Transaction
+
+The final transaction can be constructed from the unsigned transaction
+produced in <<transaction-construction>>, the signature of that transaction's
+sighash from <<sighash-construction>>, and the public key of the signing group.
+The sample function below shows the various inputs as well as how they can
+be combined to produce the final transaction hex string that can then be
+submitted to the chain:
+
+```go
+func BuildSignedTransactionHexString(
+  unsignedTransaction *wire.MsgTx, // <1>
+	signature *ecdsa.Signature, // <2>
+	publicKey *ecdsa.PublicKey, // <3>
+) (string, error) {
+  // For safety's sake, work on a deep copy, as mutations follow.
+  signedTransaction := unsignedTransaction.Copy()
+
+	// Construct witness script.
+	signatureDERBytes := (&btcec.Signature{R: signature.R, S: signature.S}).Serialize()
+	hashType := uint8(txscript.SigHashAll)
+
+	// The witness signature field is the DER signature followed by the hash type.
+	signatureForWitnessBuffer := &bytes.Buffer{}
+	signatureForWitnessBuffer.Write(signatureDERBytes)
+	binary.Write(signatureForWitnessBuffer, binary.LittleEndian, hashType)
+
+  // The witness is for the first input, since this is known to be a
+  // single-input transaction.
+  signedTransaction.TxIn[0].Witness = [][]byte{
+    signatureForWitnessBuffer.Bytes(),
+    compressedPublicKeyBytes((publicKey)),
```

Comment:
what does `compressedPublicKeyBytes` refer to?

Reply:
The tweak to witness construction that I did based on [Kuba's note](https://github.com/keep-network/keep-ecdsa/pull/670#discussion_r577527529) substitutes a real implementation here. I think I initially thought this function would need to be written separately, but never wrote it.